### PR TITLE
Check if binary is latest version

### LINKED ISSUE #988: Check for the latest version
Would be useful for the end user to be notified if there is a new version, so they get the latest features and fixes

### PR DESCRIPTION
**Pull request by lescuer97**
Needs to be decided if this is going to be used for the ASB too and where in the code base the function should be put. 

I'll wait for comments ;)

Fixes #988

### CODE REVIEW

File: swap/src/bin/swap.rs [hunk 1]
```diff
@@ -629,10 +692,43 @@ where
     Ok((btc_swap_amount, fees))
 }
 
+pub async fn check_latest_version() -> Result<()> {
+    const GITHUB_LATEST_VERSION_URL: &str =
+        "https://github.com/comit-network/xmr-btc-swap/releases/latest";
+
+    let response = match reqwest::get(GITHUB_LATEST_VERSION_URL).await {
+        Ok(res) => res,
+        Err(_) => bail!(
```
Comment (delta1) [inline swap/src/bin/swap.rs:701]: I don't think it should ever bail if it can't check for an update.

Comment (lescuer97) [reply swap/src/bin/swap.rs:701]: should I catch and print the exception inside the function instead that calling?

Comment (delta1) [reply swap/src/bin/swap.rs:701]: oh sorry! you are actually matching on the error and logging at the callsite so I think that's fine 👍

**Comment by binarybaron** [comment]
Thanks for submitting this. I think it's a reasonable feature to have. Can you rebase on master? the tests will fail otherwise

File: swap/src/bin/swap.rs [hunk 2]
```diff
@@ -67,6 +67,12 @@ async fn main() -> Result<()> {
             let swap_id = Uuid::new_v4();
 
             cli::tracing::init(debug, json, data_dir.join("logs"), Some(swap_id))?;
+
+            match check_latest_version().await {
```
Comment (binarybaron) [inline swap/src/bin/swap.rs:71]: Couldn't we match and print this error in the `check_latest_version` function? We're not using the err for anything else and that would avoid some of this duplication.

**Review by binarybaron — COMMENTED** [review]
LGTM but left one comment

File: swap/src/bin/swap.rs (hunk 2 above)
Comment (lescuer97) [reply swap/src/bin/swap.rs:71]: I'll make a note to change this and I will see how it looks. Thank you!

**Comment by binarybaron** [comment]
bors try

**Comment by lescuer97** [comment]
bors try

**Comment by delta1** [comment]
Hey @lescuer97 i think this could be DRYed up a bit by doing the check in one place per binary before matching on the command? 

If you have time to try that please give it a go otherwise I’ll add a commit later today

**Comment by lescuer97** [comment]
> Hey @lescuer97 i think this could be DRYed up a bit by doing the check in one place per binary before matching on the command?
> 
> If you have time to try that please give it a go otherwise I’ll add a commit later today

The reason this action gets repeated a lot, is because the loggin tracing gets activated inside the commands if I excecute it before: 
`cli::tracing::init`

it will not print anything, we need to have just one initialization, that could be posible but that is just another issue

**Comment by delta1** [comment]
Pushed an update @lescuer97 thanks for your work on this!

**Comment by delta1** [comment]
bors merge

**Pull request merged**